indexing-service/src/test/java/org/apache/druid/indexing/common/task/KillUnusedSegmentsTaskTest.java

```diff
@@ -1197,4 +1173,13 @@ private static DataSegment newSegment(Interval interval, String version, Map<Str
         10L
     );
   }
+
+  private void updateUsedStatusLastUpdated(DataSegment segment, DateTime newValue)
```

Comment:
This function is actually copy-pasted and used in 4 different test classes. Maybe we can find a common place we can put this in?

Reply:
Only place I can think is `TestDerbyConnector` itself, or a wrapper around it.

I would prefer a wrapper as `TestDerbyConnector` itself is completely agnostic to any column or table names and it is clean that way.
If you do end up writing a wrapper, may be the new `updateSegmentsTable` should also live there then.

Reply:
Edit: The wrapper class can be invoked via `TestDerbyConnector` itself so that any test using `TestDerbyConnector` maybe able to use it without having to explicitly declare the wrapper. So something like the following.

```
derbyConnector.segments().update("ALTER TABLE ....");

derbyConnector.segments().updateUsedStatusLastUpdated("UPDATE %s SET column=value");
```

Reply:
I like this suggestion. I went ahead with the wrapper implementation and updated usages!